app/routes.py

```diff
@@ -1,2 +1,118 @@
-from flask import Blueprint
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, make_response, request, abort
+from sqlalchemy.exc import DataError
 
+planet_bp = Blueprint("planet", __name__, url_prefix="/planets")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        message = f"{cls.__name__} {model_id} invalid"
+        abort(make_response({"message": message}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        message = f"{cls.__name__} {model_id} not found"
+        abort(make_response({"message": message}, 404))
+
+    return model
+
+@planet_bp.route("", methods=["POST"])
+def create_planet():
+    request_body = request.get_json()
+
+    try:
+        new_planet = Planet.from_dict(request_body)
+        db.session.add(new_planet)
+        db.session.commit()
+
+        message = f"Planet {new_planet.name} successfully created"
+        return make_response(jsonify(message)), 201
+    except KeyError as e:
+        expected_keys = ["name", "description", "diameter"]
+        missing_keys = [key for key in expected_keys if key not in request_body]
+
+        message = 'missing required values'
+        error_dict = {"message": message, "missing_values": missing_keys}
+
+        abort(make_response(error_dict, 400))
+    except (TypeError, DataError) as e:
+        error_dict = {
+            "message": "Invalid request body",
+            "error": str(e)
+            }
+        abort(make_response(jsonify(error_dict), 400))
```

Comment:
I love the error handling y'all did here! Great idea making sure the appropriate keys are included in the request. As mentioned before, if you are going to include this error handling, it's a good idea to be consistent and make the class' attributes non-nullable!